File: 9th lab/BitStuffing.py
'''l = []
s = int(input("Enter the size of list"))
for i in range(s):
    bs = int(input("Enter the bit streams's bit"))
    l.append(bs)
count =0
for i in range(s)'''

bitstring=input("Enter bit: ")
data=list(bitstring)
i=0
c=0
while(i<len(data)):
    if(data[i]=='1'):
        c+=1
    else:
        c=0
    if(c==6):
        data.insert(i,'0')
        c=0
    i+=1

x=''
x=''.join(data)
print(f"After bit stuffing: {x}")

File: 9th lab/BitStuffing.cpp
#include<iostream>
#include<vector>
using namespace std;
int main(){
    vector<int> v;
    int ele;
    cout<<"Enter the size of bit stream: ";
    int n;
    cin>>n;
    for(int i=0;i<n;i++){
        int ele;
        cout<<"Enter the bit to be stuffed: ";
        cin>>ele;
        v.push_back(ele);
    }
    int maxi;
    cout<<"Enter the maximum bit after which you want to stuff bit: ";
    cin>>maxi;
    int count =0;
    for(int i=0;i<v.size();i++){
        if(v[i]==1){
            count++;
        }
        else{
            count=0;
        }
        auto pos = v.begin() +i;
        if(count==6){
            v.insert(pos,'0');
        }
    }
    for(int x: v){
        cout<<x;
    }
}
